public/app/features/commandPalette/KBarResults.tsx

```diff
@@ -0,0 +1,226 @@
+    (ev: React.MouseEvent, item: RenderParams['item']) => {
+      if (typeof item === 'string') {
+        return;
+      }
+
+      // ActionImpl constructor copies all properties from action onto ActionImpl
+      // so our url property is secretly there, but completely untyped
+      // Preferably this change is upstreamed and ActionImpl has this
+      // eslint-disable-next-line
+      const url = (item as ActionImpl & { url?: string }).url;
+
+      if (item.command) {
+        item.command.perform(item);
+        query.toggle();
+      } else if (url) {
+        if (!(ev.ctrlKey || ev.metaKey || ev.shiftKey)) {
+          query.toggle();
+        }
+      } else {
+        query.setSearch('');
+        query.setCurrentRootAction(item.id);
+      }
+
+      options.callbacks?.onSelectAction?.(item);
+    },
+    [query, options]
+  );
+
+  const pointerMoved = usePointerMovedSinceMount();
+
+  return (
+    <div
+      ref={parentRef}
+      style={{
+        maxHeight: props.maxHeight || 400,
+        position: 'relative',
+        overflow: 'auto',
+      }}
+    >
+      <div
+        role="listbox"
+        id={KBAR_LISTBOX}
+        style={{
+          height: `${rowVirtualizer.totalSize}px`,
+          width: '100%',
+        }}
+      >
+        {rowVirtualizer.virtualItems.map((virtualRow) => {
+          const item = itemsRef.current[virtualRow.index];
+
+          // ActionImpl constructor copies all properties from action onto ActionImpl
+          // so our url property is secretly there, but completely untyped
+          // Preferably this change is upstreamed and ActionImpl has this
+          // eslint-disable-next-line
+          const url = (item as ActionImpl & { url?: string }).url;
+
+          const handlers = typeof item !== 'string' && {
+            onPointerMove: () =>
+              pointerMoved && activeIndex !== virtualRow.index && query.setActiveIndex(virtualRow.index),
+            onPointerDown: () => query.setActiveIndex(virtualRow.index),
+            onClick: (ev: React.MouseEvent) => execute(ev, item),
+          };
+          const active = virtualRow.index === activeIndex;
+
+          const childProps = {
+            id: getListboxItemId(virtualRow.index),
+            role: 'option',
+            'aria-selected': active,
+            style: {
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              width: '100%',
+              transform: `translateY(${virtualRow.start}px)`,
+            } as const,
+            ...handlers,
+          };
+
+          const renderedItem = React.cloneElement(
+            props.onRender({
+              item,
+              active,
+            }),
+            {
+              ref: virtualRow.measureRef,
+            }
+          );
+
+          if (url) {
+            return (
+              <a
+                key={virtualRow.index}
+                href={url}
+                // eslint-disable-next-line
```

Comment:
think we should be explicit about the rule we're ignoring here
```suggestion
          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```